src/transformers/trainer.py

```diff
@@ -1170,6 +1177,63 @@ def get_optimizer_cls_and_kwargs(args: TrainingArguments) -> Tuple[Any, Any]:
             optimizer_cls = torch.optim.Adagrad
         elif args.optim == OptimizerNames.RMSPROP:
             optimizer_cls = torch.optim.RMSprop
+        elif args.optim in [
+            OptimizerNames.GALORE_ADAMW,
+            OptimizerNames.GALORE_ADAMW_8BIT,
+            OptimizerNames.GALORE_ADAFACTOR,
+        ]:
+            from galore_torch import GaLoreAdafactor, GaLoreAdamW, GaLoreAdamW8bit
```

Comment:
We need an import check here, no? 😄

Reply:
yeah indeed, many things can be optimized, for now it's a really rough draft, will focus on polishing everything next!

Reply:
Great, ping me when you're ready for a full review :)